### PR TITLE
feat: amazon managed prometheus

### PR DESCRIPTION
**Pull request by swibrow**
## Description

Improve logging, monitoring and alerting.

Add Managed Prom option
Add pagerduty config option (fairly static)
Add alertmanager templates
Moved to built in fluent bit configs and added grep as additional filter with prefix of z to keep it as the last filter
Fluent log annotation logic can be reversed so all logs can be shipped and the exclude annotation can be used
Ability to disable addons

Small general fixes

## Motivation and Context
Implements configured state for prom stack that includes monitoring pipelines for infra. These are not required and can be disabled.

This is not the final state and more to test the prom / alert manager pipeline as a starting point.

## Breaking Changes

Removed enable_karpenter (not used)
## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,3 +1,18 @@
+variable "metadata" {
+  description = "Metadata for the platform"
+  type = object({
+    environment = optional(string, "")
+    team        = optional(string, "")
+  })
+  default = {}
+}
```
Comment (swibrow) [inline variables.tf:8]: a proposal.

File: variables.tf [hunk 2]
```diff
@@ -181,9 +211,9 @@ variable "fluent_operator" {
 }
 
 variable "fluent_log_annotation" {
-  description = "Annotation to add to pods to get logs stored in cloudwatch"
+  description = "Pod Annotation required to enable fluent bit logging. Setting name to empty string will disable annotation requirement."
   type = object({
-    name  = optional(string, "kaas.tamedia.ch/logging")
+    name  = optional(string, "fluentbit.io/include")
```
Comment (swibrow) [inline variables.tf:216]: since the default exlude annotation is fluentbit.io/exclude

File: addons.tf [hunk 3]
```diff
@@ -69,25 +73,31 @@ module "addons" {
...
     role_name_prefix = false
     # race condition if this is not disabled. Serivce type LB will use intree controller.
     # This just means annotations are needed for the service to use the aws load balancer controller
     set = [{
       name  = "enableServiceMutatorWebhook"
-      value = "false"
+      value = "true"
+      }, {
+      name  = "serviceMutatorWebhookConfig"
+      value = "Ignore"
```
Comment (swibrow) [inline addons.tf:87]: issue where aws lb controller is not ready and other deployments fail

**Review by AlfGot — APPROVED** [review]

File: examples/disable-addons/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  description = "eks module outputs"
+  value       = module.k8s_platform.eks
+}
+
+output "vpc" {
+  description = "vpc module outputs"
+  value       = module.k8s_platform.network
+}
+
+output "zconfigure_kubectl" {
```
[the remark this answers is not in the document]
Comment (swibrow) [reply examples/disable-addons/outputs.tf:11]: nope, its so its at the end of the list :)

**Pull request merged**